src/ValueObject/FileValueObject.php

```diff
@@ -0,0 +1,255 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_theme\ValueObject;
+
+use Drupal\file\Entity\File;
+use Drupal\file\FileInterface;
+use Drupal\oe_theme\ValueObject\Exception\ValueObjectFactoryException;
+
+/**
+ * FileType value object for pattern field type "file".
+ */
+class FileValueObject implements ValueObjectInterface {
+
+  /**
+   * The name of the file.
+   *
+   * @var string
+   */
+  protected $name;
+
+  /**
+   * File URL.
+   *
+   * @var string
+   */
+  protected $url;
+
+  /**
+   * The file mime type.
+   *
+   * @var string
+   */
+  protected $mime;
+
+  /**
+   * The size of the file.
+   *
+   * @var string
+   */
+  protected $size;
+
+  /**
+   * File name extension.
+   *
+   * @var string
+   */
+  protected $extension;
+
+  /**
+   * File title.
+   *
+   * @var string
+   */
+  protected $title;
+
+  /**
+   * Language code.
+   *
+   * @var string
+   */
+  protected $languageCode = '';
+
+  /**
+   * FileType constructor.
+   *
+   * @param string $name
+   *   Name of the file, e.g. "document.pdf".
+   * @param string $url
+   *   File URL, including Drupal schema if internal.
+   * @param string $mime
+   *   File mime type.
+   * @param string $size
+   *   File size in bytes.
+   */
+  private function __construct(string $name, string $url, string $mime, string $size) {
+    $this->name = $name;
+    $this->url = $url;
+    $this->mime = $mime;
+    $this->size = $size;
+  }
+
+  /**
+   * Construct object from a Drupal file entity.
+   *
+   * @param \Drupal\file\FileInterface $file_entity
+   *   Drupal file entity object.
+   *
+   * @return $this
+   */
+  public static function fromFileEntity(FileInterface $file_entity): FileValueObject {
+    $file = new static(
+      $file_entity->getFilename(),
+      $file_entity->getFileUri(),
+      $file_entity->getMimeType(),
+      (string) $file_entity->getSize()
+    );
+
+    return $file;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function fromArray(array $values = []): ValueObjectInterface {
+    $file = new static(
+      $values['name'],
+      $values['url'],
+      $values['mime'],
+      $values['size']
+    );
+
+    if (isset($values['title'])) {
+      $file->setTitle($values['title']);
+    }
+
+    if (isset($values['language_code'])) {
+      $file->setLanguageCode($values['language_code']);
+    }
+
+    return $file;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function fromAny($value): ValueObjectInterface {
```

Comment:
We have been discussing with other team members and we all kind of felt uneasy at a certain point about this. Should we remove this generic constructor? For the moment let's just use the from array one in the preprocess when in a preview context. Make sure also documentation is updated. We can leave the tests for `fromFileEntity` as we will probably use that to style media entities soon enough.